src/port1.0/portinstall.tcl

```diff
@@ -314,6 +314,14 @@ proc portinstall::create_archive {location archive.type} {
     puts $fd "@cxx_stdlib_overridden ${portinstall::cxx_stdlib_overridden}"
     close $fd
 
+    # Set standardised file modification times
+    fs-traverse file $destpath {
+        # Keep the original times for files that have not been modified during the build
+        if {[file mtime $file] > $source_date_epoch} {
+            file mtime $file $source_date_epoch
+        } 
+    }
```

Comment:
I'm not sure that will work for all ports as we expect it to. As far as I recall, Python scripts check that the timestamps of their associated .pyc files is strictly larger than the timestamp of the script itself.

For python scripts generated during the build (e.g., because of some autotools replacement magic), this would mean that we might end up shipping a pre-compiled .pyc file that ends up being ignored by all users.

Maybe we should provide an option to turn this behavior off for ports where it would have adverse effects, or add a method to override the time for certain files?

Reply:
Sorry it's taken me this long to get to this @neverpanic. Appreciate the feedback. 

> As far as I recall, Python scripts check that the timestamps of their associated .pyc files is strictly larger than the timestamp of the script itself.

Just looking it up, it seems like it should be fine for all supported python versions (see [PEP 552](https://peps.python.org/pep-0552/)). Setting SOURCE_DATE_EPOCH triggers using hash-based pyc files (https://github.com/python/cpython/commit/ccbe5818af20f8c12043f5c30c277a74714405e0).

However, I completely agree that there might be other scenarios where this has adverse effects.

> Maybe we should provide an option to turn this behavior off for ports where it would have adverse effects, or add a method to override the time for certain files?

I've added a `reproducible_timestamps` option. It's default is set to true, but this can be changed if that's the general consensus.